b42_prepush_check.sh

```diff
@@ -29,27 +34,30 @@ summary_status="${summary_status}.${uncommited_status}"
 
 echo "== Filename to function name"
 not_found=0
+OLDIFS="${IFS}"
+IFS=$'\n'
 for fname in $(find . -name 'ft_*.c'); do
-	func_name=$(basename -s .c $fname);
+	func_name="$(basename -- "$fname" .c)"
 	# grep "^/.*$func_name" "$fname"	# Check for fname in header
-	egrep -H "^(char|int|void)\s+\*?$func_name\(" $fname # Check for the function declaration
+	grep -EH "^(char|int|void|long)\s+\*+${func_name}\(" "$fname" # Check for the function declaration
```

Comment:
```suggestion
	grep -EH "^(t_[[:alnum:]_]+|(struct|enum|union)\s+[[:alnum:]_]+|unsigned|(unsigned\s+)?(char|short(\s+int)?|int|void|long(\s+long)?(\s+int)?))\s+\**${func_name}\s*\(" "$fname" # Check for the function declaration
```

Reply:
This regex will now match more return types, e.g.:
```c
int foo();
unsigned foo();
unsigned int foo();
unsigned long long int foo  ();
short int foo();
short foo();
long foo();
t_mytype   foo();
struct timeval foo();
enum e_fancy foo();
char ***foo();
unsigned void *foo     ();
...

Reply:
Looks good.
While my idea was and is: to keep scripts simple and readable, so Pisciners could read it, learn from it and contribute to it. And complex regexps make it harder.
What I think right now:
Lets merge with that commit and included, and can you after that create an issue "to document / refactor that regexp for better readability"?
I have no idea right now what it should be "better readability", but issue should be good place to discuss options what it should be and how we can make it.